fe/fe-core/src/main/java/org/apache/doris/analysis/ShowDataSkewStmt.java

```diff
@@ -58,9 +63,23 @@ public void analyze(Analyzer analyzer) throws UserException {
                     ConnectContext.get().getRemoteIP(),
                     tblRef.getName().getDb() + "." + tblRef.getName().getTbl());
         }
+
+        Optional<DatabaseIf> db = Env.getCurrentEnv().getCurrentCatalog().getDb(tblRef.getName().getDb());
+        if (!db.isPresent()) {
+            throw new AnalysisException("Can not find db: " + tblRef.getName().getDb());
+        }
+        Optional<TableIf> tbl = db.get().getTable(tblRef.getName().getTbl());
+        if (!tbl.isPresent()) {
+            throw new AnalysisException("Can not find table: " + tblRef.getName().getDb() + "."
+                    + tblRef.getName().getTbl());
+        }
+
         PartitionNames partitionNames = tblRef.getPartitionNames();
-        if (partitionNames == null || partitionNames.getPartitionNames().size() != 1) {
-            throw new AnalysisException("Should specify one and only one partition");
+        if (partitionNames == null) {
+            if (tbl.get().getPartitionNames().isEmpty()) {
+                throw new AnalysisException("Can not find any partition");
+            }
+            tblRef.setPartitionNames(new PartitionNames(false, new ArrayList<>(tbl.get().getPartitionNames())));
```

Comment:
do you think we should we also show tempPartitions? getPartitionNames() does not handle tempPartitions.

Reply:
> do you think we should we also show tempPartitions? getPartitionNames() does not handle tempPartitions.

@rohitrs1983  OK, I will add both later.